src/Nethermind/Nethermind.HealthChecks/FreeDiskSpaceChecker.cs

```diff
@@ -0,0 +1,111 @@
+//  Copyright (c) 2022 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+//
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Nethermind.Api;
+using Nethermind.Core.MessageBus;
+using Nethermind.Logging;
+
+namespace Nethermind.HealthChecks
+{
+    public readonly struct LowDiskSpaceMessage : IMessage
+    {
+        public LowDiskSpaceMessage(long freeDiskSpace)
+        {
+            AvailableDiskSpce = freeDiskSpace;
+        }
+        public long AvailableDiskSpce { get; init; }
+    }
+
+    internal class FreeDiskSpaceChecker : IHostedService, IAsyncDisposable
+    {
+        private readonly ISimpleMessageBus _messageBus;
+        private readonly IHealthChecksConfig _healthChecksConfig;
+        private readonly IInitConfig _initConfig;
+        private readonly ILogger _logger;
+        private readonly PeriodicTimer _timer;
+        private Task _timerTask;
+        public static readonly int BytesToGB = 1024 << 20;
+
+        public FreeDiskSpaceChecker(INethermindApi api)
```

Comment:
NP: Recommend passing in individual component. Makes writing unit test easier. And forces you to think again if you have too many dependencies.

Reply:
Yes, sounds good - will change.